zio-http/src/test/scala/zhttp/internal/AppPort.scala

```diff
@@ -0,0 +1,24 @@
+package zhttp.internal
+
+import zio._
+
+object AppPort {
+  sealed trait Service {
+    def set(n: Int): IO[Nothing, Unit]
+    def get: IO[Nothing, Int]
+  }
+
+  def live: ZLayer[Any, Nothing, AppPort] = Ref
+    .make(0)
+    .map(ref => new Live(ref))
+    .toLayer
+
+  case class Live(ref: Ref[Int]) extends Service {
+    override def set(n: Int): IO[Nothing, Unit] = ref.set(n)
+
+    override def get: IO[Nothing, Int] = ref.get
+  }
+
+  def set(n: Int): ZIO[AppPort, Nothing, Unit] = ZIO.accessM(_.get.set(n))
+  def get: ZIO[AppPort, Nothing, Int]          = ZIO.accessM[AppPort](_.get.get)
```

Comment:
This is dangerous. The call to `get` might return `0` if not already set. I would prefer using a promise instead of a Ref over here.

Also - U can move this logic to `AppCollection` class. We can rename that class to something else.

Reply:
But won't ref ensure an atomic operation in the over-all effect? What I mean is, the first operation effectively defines a server start up(sets up the value) and then subsequent operations get the value.
Won't set take precedence in this case everytime because of how ref is designed?